cpp/ps0007/problem_s_8.cpp

```diff
@@ -39,17 +42,22 @@ int main() {
         paramCount = 1;
     } else {
         cout << "Shape not recognized." << endl;
-        return 0;
+        return 1;
     }
-
-    unsigned params[paramCount];
+    
+    // fill in the dimensions array 2/3
+    float params[paramCount];
     for (int i = 0; i < paramCount; ++i) {
         cout << "Enter parameter " << i + 1 << ": ";
         cin >> params[i];
+        if (params[i] <= 0)
+            i--;
     }
 
+    // printing out the area of the shape 1/1
     float area = calculateArea(shape, params);
     cout << "The area of the " << shape << " is: " << area << endl;
 
     return 0;
-}
\ No newline at end of file
+}
+// great work 😎 8/10 ⭐
```

Comment:
- [x] program is in the correct path
- [ ] no syntax errors while compilation & no bugs or warnings while execution
- [x] formatted code to be clear and readable
- [ ] commented some phrases for blocks of code
- [x] used array & function concepts for the program
- [ ] good design for function using DRY concept
- [ ] main returns non-zero/zero number according to the case
+0.5 bonus mark for your effort ⭐
+0.5 bonus mark for validating more shapes area ⭐

you can check the [problem solution](https://onlinegdb.com/wybZ67yJX) for more